### PR TITLE
#154953732 Allow creation of users via REST API 

### PR DESCRIPTION
**Pull request by Sebuliba-Adrian**
#### What does this PR do?
Allow creation of users via REST API 
#### Description of Task to be completed?
* Allows  creating new users via the API. External apps might need to register 
accounts on behalf of their users without having 
to send them over to the wger instance. 
* Apps register as a regular user to get an API key
By default, users aren't allowed to create new accounts over the API, this has to be explicitly allowed

#### How should this be manually tested?
* execute python manage.py runserver
* A registered app can create users with the following command:
     python manage.py add-user-rest-api <app_name>
#### What are the relevant pivotal tracker stories?
  [154953732](https://www.pivotaltracker.com/story/show/154953732)

### CODE REVIEW

**Comment by Sebuliba-Adrian** [comment]
@collin5 please review this PR. Thank you

File: wger/core/tests/test_registration.py [hunk 1]
```diff
@@ -137,3 +140,61 @@ def test_registration_deactivated(self):
...
+        user_profile.save()
+
+        reg_data = dict(username='test_user',
+                        password='test_password', email='test@mail.com')
+
+        self.user_login()  # this user's flag can_add_user is set to False
+        response = self.client.post(url, data=reg_data)
+        self.assertEqual(response.status_code, 403)
+
+        # Test register via Rest API
```
Comment (collin5) [inline wger/core/tests/test_registration.py]: Could you separate this test to another function

File: wger/core/tests/test_registration.py [hunk 2]
```diff
@@ -137,3 +140,61 @@ def test_registration_deactivated(self):
...
+        self.new_user_login()  # this user's flag can_add_user is set to True
+        response = self.client.post(url, data=reg_data)
+        reply = json.loads(response.content.decode('utf-8'))
+        self.assertEqual(reply['Message'],
+                         "Profile created", msg="Profile not created")
+
+        # Test username exists
```
Comment (collin5) [inline wger/core/tests/test_registration.py]: Same with this one

File: wger/core/tests/test_registration.py [hunk 3]
```diff
@@ -137,3 +140,61 @@ def test_registration_deactivated(self):
...
+        response_post = self.client.post(url, data=reg_data)
+        self.assertEqual(response_post.status_code, 400)
+
+        # Test incomplete data
```
Comment (collin5) [inline wger/core/tests/test_registration.py]: This too

File: wger/core/tests/test_registration.py [hunk 4]
```diff
@@ -137,3 +140,61 @@ def test_registration_deactivated(self):
...
+        reg_data.pop('password')
+        response_post = self.client.post(url, data=reg_data)
+        self.assertEqual(response_post.status_code, 400)
+
+        # Invalid data missing username
```
Comment (collin5) [inline wger/core/tests/test_registration.py]: Same with this

**Comment by Sebuliba-Adrian** [comment]
@collin5 Thanks for the feedback. Let me separate the tests into a separate function each.

**Comment by Sebuliba-Adrian** [comment]
@collin5 Please review this PR. The changes you requested for have been made. Thank you.

**Comment by Sebuliba-Adrian** [comment]
@collin5 I have applied all the changes as requested. Please review this PR. Thank you

**Pull request merged**